lib/patchelf/patcher.rb

```diff
@@ -20,12 +20,25 @@ class Patcher
     #   Filename of input ELF.
     # @param [Boolean] logging
     #   Whether to use a (stderr-initialized) logger for errors
-    def initialize(filename, logging: true)
+    # @param [:log, :silent, :exception] on_error
+    #   :log = logs to stderr
+    #   :exception = raise exception related to the error
+    #   :silent = ignore the errors
+    #
+    # @raise [ArgumentError] when both logging and on_error are passed to initializer
+    def initialize(filename, logging: nil, on_error: nil)
```

Comment:
I know you want to detect the scenario that both `logging` and `on_error` are passed, but this prototype makes it unclear that default behavior is `on_error: :log`.
I would prefer declare `on_error: :log` and just ignore the scenario. We only need it be compatible to old prototypes, not really need foolproof.

(And if you adopt this opinion, we can keep `logging` default to `false` as well)

Reply:
i kept `on_error: :log` , `logging: nil` by default.
This won't capture when `on_error: :log, logging: any_value` is passed.

we can work around this by accepting hash as argument,  idk if that's worth the effort.

Reply:
I found my previous comment was wrong, it should be "we can keep `logging` default to `true` as well"